src/libcore/iter/iterator.rs

```diff
@@ -2463,6 +2463,52 @@ pub trait Iterator {
             }
         }
     }
+
+    /// Extends passed collection with the contents of an iterator.
+    ///
+    /// This method consumes the iterator and adds all its items to the
```

Comment:
This phrasing seems to indicate that `extended` will gain `self.count()` number of new elements while it may gain zero new elements in the case of sets. I'd rephrase `add` to `include`.

Reply:
It might also be prudent to mention that `.by_ref()` exists if people are using `.take(..)` and such things. This also applies to `extend_mut`.